predicates/auth/jwt.go

```diff
@@ -0,0 +1,183 @@
+/*
+Package auth implements custom predicates to match based on content
+of the HTTP Authorization header.
+
+You can use this predicate to match a route based on data in the 2nd part of a JWT token, for example based on the issuer.
+
+Examples:
+
+    // one key value pair has to match
+    example1: JWTPayloadAnyKV("iss", "https://accounts.google.com", "email", "skipper-router@googlegroups.com")
+	-> "http://example.org/";
+    // all key value pairs have to match
+    example2: * && JWTPayloadAllKV("iss", "https://accounts.google.com", "email", "skipper-router@googlegroups.com")
+	-> "http://example.org/";
+*/
+package auth
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/zalando/skipper/routing"
+)
+
+const (
+	authHeaderName   = "Authorization"
+	authHeaderPrefix = "Bearer "
+)
+
+var InvalidArgsError = errors.New("invalid arguments")
```

Comment:
errors should be called conventionally ErrSomething. -> ErrInvalidArgsError. For some reason we also have this error here: https://github.com/zalando/skipper/blob/master/predicates/predicates.go#L6